### ISSUE TITLE
Improve Eslint Rules

### ISSUE THREAD

**Issue by palashgdev** [SigNoz/signoz#409]
## Bug description

*Please describe.*  
*If this affects the front-end, screenshots would be of great help.*  
Current eslint config allow array index as key which is not a good solution

## Expected behavior

Eslint rules should be more strict

**Comment by pal-sig** [comment]
As @jyash97 pointed out we should not be using index as key 

probably more strict eslint rules would help here

**Comment by pranay01** [comment]
@pal-sig Can you add more details on what is expected here? What specific way should eslint rules be made stricter?

**Comment by pal-sig** [comment]
There is good ans for this [issue](https://stackoverflow.com/a/46735689)

**Comment by genzyy** [comment]
Can I take up this issue? I might be able to fix this with `eslint-plugin-react` as it provides an eslint rule for this. Or if this can be done in any better way then can you tell me that so I can work on this?

**Comment by pranay01** [comment]
ok, assigning this to you @genzyy 

@pal-sig Please guide @genzyy

**Comment by genzyy** [comment]
@pal-sig How should I go with solving this issue?
I think adding the rule in `.eslintrc.cjs` using [this](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md) should do it or am I missing something here?

**Comment by pal-sig** [comment]
> @pal-sig How should I go with solving this issue? I think adding the rule in `.eslintrc.cjs` using [this](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md) should do it or am I missing something here?

Yes @genzyy

**Comment by genzyy** [comment]
I have added the rule for this [here](https://github.com/genzyy/signoz/blob/unallow-array-index-key/frontend/.eslintrc.cjs). If any changes are required let me know. I have created a branch for this while keeping main as the base.

**Comment by pal-sig** [comment]
> I have added the rule for this [here](https://github.com/genzyy/signoz/blob/unallow-array-index-key/frontend/.eslintrc.cjs). If any changes are required let me know. I have created a branch for this while keeping main as the base.

Yup this will work... also would it be possible ... to fix those line in the codebase (if any) which are affected by this rule ?

**Comment by genzyy** [comment]
> Yup this will work... also would it be possible ... to fix those line in the codebase (if any) which are affected by this rule ?

Yeah, let me check those errors first.

**Comment by genzyy** [comment]
Yup I got some errors because of the new `eslint` rule that I just added. How do you want to fix those errors like do you have any specific way of doing it (setting a unique key )?

I thought of two ways:

- use something like `uuid` to create unique numbers.
- use of `date` or `time` and concatenate with index.

**Comment by pal-sig** [comment]
Hi @genzyy 

both ways are incorrect use array data only rather than array's index

**Comment by genzyy** [comment]
> Hi @genzyy
> 
> both ways are incorrect use array data only rather than array's index

Okay got it.
Also I have seen in many components, for example in `TimePreferenceDropdown`, no key value is being passed when rendering components using map function. So should I add a key to those components as well?

**Comment by genzyy** [comment]
> > I have added the rule for this [here](https://github.com/genzyy/signoz/blob/unallow-array-index-key/frontend/.eslintrc.cjs). If any changes are required let me know. I have created a branch for this while keeping main as the base.
> 
> Yup this will work... also would it be possible ... to fix those line in the codebase (if any) which are affected by this rule ?

I have fixed them. I ll create a new pull request now.

**Comment by genzyy** [comment]
Hey can I work on this issue again?  I made the original PR for fixing this issue so maybe I will be able to fix it.

**Comment by pal-sig** [comment]
> Hey can I work on this issue again? I made the original PR for fixing this issue so I will be able to fix it.

Hi 
If you can improve eslint config please raise a pr
